server/src/main/java/org/apache/iotdb/db/engine/snapshot/SnapshotLoader.java

```diff
@@ -217,142 +224,203 @@ private void deleteAllFilesInDataDirs() throws IOException {
-            String[] splitPath =
-                unseqPartitionDir
-                    .getAbsolutePath()
-                    .split(File.separator.equals("\\") ? "\\\\" : File.separator);
-            long timePartition = Long.parseLong(splitPath[splitPath.length - 1]);
-            File[] files = unseqPartitionDir.listFiles();
-            if (files != null && files.length > 0) {
-              Arrays.sort(files, Comparator.comparing(File::getName));
-              String currDir = DirectoryManager.getInstance().getNextFolderForUnSequenceFile();
-              for (File file : files) {
-                if (file.getName().endsWith(".tsfile")) {
-                  currDir = DirectoryManager.getInstance().getNextFolderForUnSequenceFile();
-                }
-                File targetFile =
-                    new File(
-                        currDir,
-                        storageGroupName
-                            + File.separator
-                            + dataRegionId
-                            + File.separator
-                            + timePartition
-                            + File.separator
-                            + file.getName());
-                if (!targetFile.getParentFile().exists() && !targetFile.getParentFile().mkdirs()) {
-                  throw new IOException(
-                      String.format("Failed to create dir %s", targetFile.getParent()));
-                }
-                try {
-                  Files.createLink(targetFile.toPath(), file.toPath());
-                } catch (FileSystemException e) {
-                  Files.copy(file.toPath(), targetFile.toPath());
-                }
-              }
-            }
-          }
-        }
+        String targetSuffix =
+            IoTDBConstant.UNSEQUENCE_FLODER_NAME
+                + File.separator
+                + storageGroupName
+                + File.separator
+                + dataRegionId
+                + File.separator
+                + timePartitionFolder.getName();
+        createLinksFromSnapshotToSourceDir(targetSuffix, files);
       }
     }
   }
 
-  private void createLinksFromSnapshotDirToDataDirWithLog() {
-    while (logAnalyzer.hasNext()) {
-      Pair<String, String> filesPath = logAnalyzer.getNextPairs();
-      File sourceFile = new File(filesPath.left);
-      File linkedFile = new File(filesPath.right);
-      if (!linkedFile.exists()) {
-        LOGGER.warn("Snapshot file {} does not exist, skip it", linkedFile);
-        continue;
-      }
-      if (!sourceFile.getParentFile().exists() && !sourceFile.getParentFile().mkdirs()) {
-        LOGGER.error("Failed to create folder {}", sourceFile.getParentFile());
-        continue;
+  private void createLinksFromSnapshotToSourceDir(String targetSuffix, File[] files)
+      throws DiskSpaceInsufficientException, IOException {
+    for (File file : files) {
+      String dataDir = folderManager.getNextFolder();
```

Comment:
What is this operation used for ? Distribute the snapshot files to different data dirs ?